src/protections/ProtectionManager.ts

```diff
@@ -81,9 +65,14 @@ class EnabledProtectionsManager extends MatrixDataManager<EnabledProtectionsEven
     private readonly enabledProtections = new Set</* protection name */string>();
 
     constructor(
-        private readonly mjolnir: Mjolnir
+        private readonly mjolnir: Mjolnir,
+        private readonly protections: Protection[]
     ) {
         super()
+        const banPropagationProtection = this.protections.find(p => p.name === 'BanPropagationProtection');
```

Comment:
would be cool if we could make 'BanPropagationProtection' a static `name` field on the BanPropagationProtection class so we don't have to risk typoing it. Which would then remove the need for this check